src/transforms/filter.jl

```diff
@@ -0,0 +1,60 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(function)
+
+Filters the table returning only the rows where the `function` returns true.
+"""
+struct Filter{F} <: Stateless
+  func::F 
+end
+
+isrevertible(::Type{<:Filter}) = true
+
+rowvalues(row, colnames) = [Tables.getcolumn(row, col) for col in colnames]
+
+function apply(transform::Filter, table)
+  tablerows = Tables.rows(table)
+  colnames = Tables.columnnames(table)
+  rows = [rowvalues(row, colnames) for row in tablerows]
+
+  # selected rows and rejected rows/inds 
+  srows = Vector{eltype(rows)}()
+  rrows = Vector{eltype(rows)}()
+  rinds = Vector{Int}()
+  
+  for (i, row) in enumerate(rows)
+    if transform.func(row)
+      push!(srows, row)
+    else
+      push!(rrows, row)
+      push!(rinds, i)
+    end
+  end
```

Comment:
This could be replaced by a findall or filter call with arrays of tuples.

Reply:
OK, I'll do it.